src/blueprint/azext_blueprint/custom.py

```diff
@@ -0,0 +1,577 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=line-too-long
+# pylint: disable=too-many-statements
+# pylint: disable=too-many-lines
+# pylint: disable=too-many-locals
+# pylint: disable=unused-argument
+
+import json
+import os
+from knack.util import CLIError
+
+
+def import_blueprint_with_artifacts(cmd,
+                                    client,
+                                    blueprint_name,
+                                    input_path,
+                                    management_group=None,
+                                    scope=None):
+    from ._client_factory import cf_artifacts
+
+    artifact_client = cf_artifacts(cmd.cli_ctx)
+    body = {}
+    blueprint_path = os.path.abspath(os.path.join(input_path, 'blueprint.json'))
+
+    with open(blueprint_path) as blueprint_file:
+        blueprint = json.load(blueprint_file)
+        if blueprint['properties'] is None:
+            raise CLIError('Blueprint file does not contain properties field')

```

Comment:
This error is confusing. If the field 'properties' does not exist, python will immediately throw an KeyError exception, and hence not hit this line which seems to want to inform the user that the field properties does not exist in the file. Again, though, it would not get hit if the key 'properties' does not exist. I think you should use the python dict get method if you want this to work properly. This will return None if the key doesn't exist, thus hitting the raise CLIError. You can test this easily by modifying one of your test .json files an see the KeyError:
  File "C:\Users\stevens\Projects\env\lib\site-packages\azext_blueprint\custom.py", line 30, in import_blueprint_with_artifacts
    if blueprint['properties'] is None:
KeyError: 'properties'

Reply:
Thanks for spotting this. I've fixed it.